classes/local/attributemapping.php

```diff
@@ -100,6 +100,8 @@ private function validate_table($table) {
      * @throws InvalidArgumentException If the field name is invalid
      */
     private function validate_field($table, $field) {
+        if (!$field) return;
```

Comment:
Allowing non-selected values as users could leave the field blank and save the form.